### PR TITLE
Add recipe for the Racket language.

### PR DESCRIPTION
**Pull request by insertinterestingnamehere**
I needed a package for this for a project recently. Hopefully others find it useful.

### CODE REVIEW

File: recipes/racket/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  url: https://github.com/racket/racket/archive/v{{ version }}.tar.gz
+  sha256: c98f8d8f8dd62abdef9906adf7b1ce60b37c852b4111700c06bf0c2d290dd588
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
```
Comment (jakirkham) [inline recipes/racket/meta.yaml]: This has changed. Please see [these docs]( https://conda-forge.org/docs/meta.html#compilers ).

File: recipes/racket/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+test:
+  commands:
+    - racket -e "\"hello world\""
+
+about:
+  home: http://racket-lang.org/
+  license: LGPL 3
```
Comment (jakirkham) [inline recipes/racket/meta.yaml]: Is there a license file we package as well?

File: recipes/racket/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Start with a minimal build without any packages.
+# Avoid building any packages since they could potentially
+# require a more recent version of the base language than
+# is in the current source directory.
+# Packages can be added once the package manager is configured
+# to work for this particular version number.
+make unix-style PREFIX=$PREFIX PKGS=""
```
Comment (jakirkham) [inline recipes/racket/build.sh]: Is there a test suite we can run (possibly even a small portion of a test suite)?

File: recipes/racket/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{% set version = "6.12" %}
+
+package:
+  name: racket
+  version: {{version}}
```
Comment (jakirkham) [inline recipes/racket/meta.yaml]: Please add a space after `{{` and before `}}`.

File: recipes/racket/build.sh (hunk 3 above)
Comment (jakirkham) [inline recipes/racket/build.sh]: Would quote `$PREFIX` in case of spaces.

Comment (jakirkham) [inline recipes/racket/build.sh]: Please set `CPUS` to be `CPU_COUNT`.

ref: https://github.com/racket/racket/blob/daa7ddeef8c40dd4f1ccd5f158219daf1cecda52/INSTALL.txt#L161

ref: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/b4163ff95b38e6d657a9be3f18caca3db3bb8b20/recipe/run_conda_forge_build_setup_linux#L3-L5

ref: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/b4163ff95b38e6d657a9be3f18caca3db3bb8b20/recipe/run_conda_forge_build_setup_osx#L3-L5

**Comment by jakirkham** [comment]
Looks pretty good. Thanks for adding this. Some minor comments above.

**Comment by insertinterestingnamehere** [comment]
Their test suite is hanging on circleci for some reason. That needs further investigation, though I'm not sure how much of that needs to be done on our end vs just reported to the racket developers.

Other than that, I've addressed the outstanding comments. Let me know if there's anything else that needs to be done there.

**Comment by insertinterestingnamehere** [comment]
Currently I have this set up to install packages from the language package manager alongside the main language installation inside the conda environment. Is that a reasonable approach?

**Comment by jakirkham** [comment]
This looks pretty good to me.

FWIW one can run the Linux tests locally as long as Docker is installed. Just run `./.circleci/run_docker_build.sh`. When the process terminates, the container remains so that it can be inspected. These accumulate after a while; so, make sure to clear them out periodically.

We could also just skip the failing test for now and save an issue about it in the feedstock.

I'm not sure about the package manager portion. This could be reasonable. OTOH there is some value to having separate packages for these things. Could also evaluate split packages in this context. Maybe others have thoughts here?

cc @conda-forge/staged-recipes

**Comment by scopatz** [comment]
It seems like this is how pip works with conda right now, in that the packages go into the conda dir.

**Comment by insertinterestingnamehere** [comment]
Okay, I've submitted the error upstream at https://github.com/racket/racket/issues/2137. For the time being, I'll disable the failing tests here.

**Comment by insertinterestingnamehere** [comment]
Yah, I think it makes sense to have the language packages install alongside the language installation since it lets both be swapped out together when the conda environment changes. Conda packages are often installed without root permissions anyway, so a user will usually be able to install whatever packages they want in their conda environments without any trouble.

**Comment by insertinterestingnamehere** [comment]
I think this is done.

**Comment by jakirkham** [comment]
Thanks @insertinterestingnamehere. Sorry for the slow reply. We have been having issues converting recipes of late. ( https://github.com/conda-forge/staged-recipes/issues/5846 ) Hence we hadn't merged this yet. Agree it looks fine. Will try merging to see if we have fixed things.

**Comment by insertinterestingnamehere** [comment]
No worries. Thank you!

**Pull request merged**